src/main/java/org/jboss/config/YamlObjectMapper.java

```diff
@@ -0,0 +1,33 @@
+package org.jboss.config;
+
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import com.fasterxml.jackson.module.paramnames.ParameterNamesModule;
+import io.quarkus.arc.All;
+import io.quarkus.jackson.ObjectMapperCustomizer;
+import jakarta.inject.Singleton;
+import jakarta.ws.rs.Produces;
+
+import java.util.List;
+
+public class YamlObjectMapper {
+    @Singleton
+    @Produces
+    ObjectMapper objectMapper(@All List<ObjectMapperCustomizer> customizers) {
+        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+        objectMapper.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.registerModule(new Jdk8Module());
+        objectMapper.registerModule(new ParameterNamesModule());
+
+        // Apply all ObjectMapperCustomizer beans (incl. Quarkus)
+        for (ObjectMapperCustomizer customizer : customizers) {
+            customizer.customize(objectMapper);
```

Comment:
I'm not an expert but should you rather define customizer then overriding the whole ObjectMapper? Can't it be used in different parts of quarkus?

Reply:
since this is used everywhere in Quarkus I'm not sure if it's good idea to override default objectmapper with yaml. Maybe rather provide your custom bean that we can use in this app.

Reply:
I think we should be fine, it was done as it is done in the quarkus github app, please see https://github.com/quarkiverse/quarkus-github-app/blob/main/runtime/src/main/java/io/quarkiverse/githubapp/runtime/UtilsProducer.java